scapy/layers/tls/automaton_cli.py

```diff
@@ -820,44 +846,73 @@ def TLS13_START(self):
     @ATMT.condition(TLS13_START)
     def tls13_should_add_ClientHello(self):
         # we have to use the legacy, plaintext TLS record here
+        supported_groups = ["secp256r1", "secp384r1", "x25519"]
         self.add_record(is_tls13=False)
         if self.client_hello:
             p = self.client_hello
         else:
-            # When trying to connect to a public TLS 1.3 server,
-            # you will most likely need to provide an SNI extension.
-            # sn = ServerName(servername="<put server name here>")
-            ext = [TLS_Ext_SupportedGroups(groups=["secp256r1"]),
-                   # TLS_Ext_ServerName(servernames=[sn]),
-                   TLS_Ext_KeyShare_CH(client_shares=[KeyShareEntry(group=23)]),  # noqa: E501
-                   TLS_Ext_SupportedVersions(versions=["TLS 1.3-d18"]),
-                   TLS_Ext_SignatureAlgorithms(sig_algs=["sha256+rsapss",
-                                                         "sha256+rsa"])]
-            p = TLSClientHello(ciphers=0x1301, ext=ext)
+            if self.ciphersuite is None:
+                c = 0x1301
+            else:
+                c = self.ciphersuite
+            p = TLS13ClientHello(ciphers=c)
+
+        ext = []
```

Comment:
Is there any reason for this change? I prefer a static list created rather than that. Also, using  a variable might be useless, and we can get rid of the `noqa:`s:
```Python
        p.ext = [
            TLS_Ext_SupportedVersion_CH(versions=["TLS 1.3"]),
            TLS_Ext_SupportedGroups(groups=supported_groups),
            TLS_Ext_KeyShare_CH(
                client_shares=[KeyShareEntry(group=self.curve)],
            ),
            TLS_Ext_SignatureAlgorithms(sig_algs=["sha256+rsaepss"]),
        ]
```

Reply:
Yes the reason for this change is because the list of extensions can't always be static if we want to implement all the TLS 1.3 handshake functionalities. For example, the extensions will change depending we are authenticating with PSK, sending 0-RTT encrypted data...

So this change is not really particularly helpful for this PR because it implements only a "basic" 1-RTT handshake but it's mainly to prepare for the next PR where the list of extensions will change. 

I will fix the noqa: and get rid of the useless variable in my next commit.

Reply:
I'd rather keep a static list for now, and add dynamic changes when needed. Would that work for you?

Reply:
I changed that in my last commit but I've made a minor modification to add the static list if there's no extension provided in self.client_hello